instructure/canvas-ios
# Course Sync Selector Screen

Course Sync Selector UI and logic. Link to figma can be found [here](https://www.figma.com/file/w9DAKBl1PbHztkBxjs9BF3/%E2%9C%85-%5BDelivered%5D-(Student)-Offline-Usage?type=design&node-id=697-58611&t=eI7FQbaG4Ww9t8t7-0). 

Test plan: 
1. Enable `offline_mode` feature flag. 
2. Tap the cloud icon in the dashboard
3. You should see your course list 
4. Enabled tab list: Assignments, Pages, Files 
5. By default everything is deselected and collapsed, sync button is disabled
6. Check if tabs are visible
7. Check if every file is visible 
8. Check full and partial selection

## Screenshots

<table>
<tr><th>Deselected</th><th>Selected</th></th><th>Partially selected</th></tr>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/110813321/203e1ec5-6cc4-480e-9d67-9aa98bf09ec8" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/110813321/95522375-7997-4115-9b4e-56a2a21b0309" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/110813321/15f83ccd-0f33-46fd-942b-a94867232557" maxHeight=500></td>

<tr><th>Dark mode Deselected</th><th>Dark mode Selected</th></th><th>Dark mode Partially selected</th></tr>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/110813321/d9f06109-e87d-40de-bc19-dfd5aacdeaa6" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/110813321/31625401-2d26-41f0-9f45-e33821acb7ab" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/110813321/b9483f82-6c36-4761-ad09-c3ce6ef69f16" maxHeight=500></td>
</tr>
</table>

## Checklist
